### PR TITLE
[M] Log changes to content access mode for owner (ENT-4121)

### PR DESCRIPTION
**Pull request by wolfdale**
Changes
- Logging changes to content access mode for owner.

### CODE REVIEW

**Review by Ceiu — CHANGES_REQUESTED** [review]
A better place for the logging would be the ContentAccessManager, which handles this logic for most(/every) means of changing the SCA mode or list.

Check ContentAccessManager:~760

**Review by nikosmoum — COMMENTED** [review]
@Ceiu I think that is called in both the create and update endpoints, and we only want to log this during the update endpoint. Also, we don't want to log it in case something goes wrong during the rest of the call, which is why @wolfdale added the logging after the flush is done

**Comment by Ceiu** [comment]
@nikosmoum Import as well, but prod likely doesn't care about that. It just seems odd to me to add logging in one specific area, given that we have a centralized place for this logic, that also kicks out a specific event when a change occurs. Not wanting it outputting on creation is weird to me -- especially given that the initial mode is configurable on creation (thus, functionally identical to the state flipping they're interested in checking).

FWIW, ContentAccessManager does its own merge on the owner. In the case where the SCA stuff changes during updateOwner, that second merge+flush is redundant.

**Comment by nikosmoum** [comment]
@Ceiu Ack, let's move this in the `ContentAccessManager` then

**Comment by nikosmoum** [comment]
@wolfdale Also, I think we need to port this to 4.0 too

**Comment by wolfdale** [comment]
@nikosmoum @Ceiu Ack.
I'll cherrypick this to 4.0 once this gets merged.

**Review by nikosmoum — APPROVED** [review]
ack from me. I'll let @Ceiu re-review & merge

File: src/main/java/org/candlepin/resource/OwnerResource.java
Comment (Ceiu) [inline src/main/java/org/candlepin/resource/OwnerResource.java:983]: I'm hesitant to okay this, since we intentionally left it off prior to avoid logical coupling to the CAM for merging the product data. With the added comment it's probably fine, as future maintainers would have a hint as to why the merge isn't always done; but in such a case we should reuse the ```updateContentAccess``` check we've already done directly above this block and drop it in as an "else" block.

**Review by Ceiu — CHANGES_REQUESTED** [review]

File: src/main/java/org/candlepin/resource/OwnerResource.java
Comment (wolfdale) [reply src/main/java/org/candlepin/resource/OwnerResource.java]: I was initially thinking of putting everything inside of else block.
Done.

**Review by Ceiu — APPROVED** [review]

**Pull request merged**
